### ISSUE TITLE
parse error when template literal type starts with `<` or `>`

### ISSUE THREAD

**Issue by codehz** [denoland/deno#11140]
poc:
```typescript
type POC<s extends string> = s extends `<` ? true : false;
```

Error:
```
Error formatting: <filename>
   Line 1, column 41: Unexpected token `<`. Expected template token

  tring> = s extends `<` ? true :
                      ~
Checked 1 file
```

**Comment by magurotuna** [comment]
This issue comes from the upstream crate(swc). I've reported it there.

**Comment by NfNitLoop** [comment]
I'm running into this issue too.  (See: [Discord conversation](https://discord.com/channels/684898665143206084/689420767620104201/860611904292388915))



When I try to `export * as h2md from "https://cdn.skypack.dev/node-html-markdown@v0.1.7?dts"`, Deno gives me the error: 

```
Download https://cdn.skypack.dev/-/domhandler@v4.2.0-u7hAtYaFZZGmyJHlmYtb/dist=es2020,mode=imports/optimized/domhandler.js
Download https://cdn.skypack.dev/-/dom-serializer@v1.3.2-5J9msNzAwNbdDUcRWyqA/dist=es2020,mode=imports/optimized/dom-serializer.js
Download https://cdn.skypack.dev/-/domelementtype@v2.2.0-gq4U3YDkWc2fpHgdKziO/dist=es2020,mode=imports/optimized/domelementtype.js
error: Unexpected token `<`. Expected template token at https://cdn.skypack.dev/-/node-html-parser@v3.3.6-4CBBorgyTawMTjHyh9V2/dist=es2020,mode=types/dist/nodes/comment.d.ts:17:17
```

Which seems to be referring to the `<` in this inside of [comment.d.ts]:

[comment.d.ts]:  https://cdn.skypack.dev/-/node-html-parser@v3.3.6-4CBBorgyTawMTjHyh9V2/dist=es2020,mode=types/dist/nodes/comment.d.ts

```typescript
export default class CommentNode extends Node {
    rawText: string;
    constructor(rawText: string, parentNode: HTMLElement);
    /**
     * Node Type declaration.
     * @type {Number}
     */
    nodeType: NodeType;
    /**
     * Get unescaped text value of current node and its children.
     * @return {string} text content
     */
    get text(): string;
    toString(): `<!--${string}-->`;
}
```

**Comment by NfNitLoop** [comment]
https://github.com/swc-project/swc/issues/1862 is the upstream bug  for others interested in following along. :)

**Comment by F3n67u** [comment]
@magurotuna bug have already fixed in swc. can we upgrade our swc version to fix this issue now?

**Comment by bartlomieju** [comment]
> @magurotuna bug have already fixed in swc. can we upgrade our swc version to fix this issue now?

Yes, it will be updated prior to 1.13 release

**Comment by bartlomieju** [comment]
Fixed in #11626
